.github/workflows/integration-cloud.yml

```diff
@@ -13,20 +13,58 @@ concurrency:
   cancel-in-progress: false
 
 jobs:
+  setup-environment:
+    runs-on: ubuntu-latest
+    outputs:
+      python-cache-key: ${{ steps.cache-python.outputs.cache-hit }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Cache dependencies
+        id: cache-python
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/pyproject.toml') }}
+
+      - name: Install dependencies
+        if: steps.cache-python.outputs.cache-hit != 'true'
+        run: |
+          pip install -e .[dev]
+
+      - name: Upload Python environment
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-environment
+          path: ~/.cache/pip
```

Comment:
I don't understand why you would use this instead of simply rely on `actions/cache` for restoring as well?